ui_testing/pages/order_page.py

```diff
@@ -258,6 +258,84 @@ def duplicate_suborder(self):
                                                                        destination_element='order:duplicate_table_view')
         duplicate_element.click()
 
+    # this method to be used while you are order's table with add page ONLY, and you can get the required data by sending the index, and the needed fields of the suborder
+    def get_suborder_data(self, sub_order_index=0, departments=True, material_type=True, articles=True, test_plan=True, test_unit=True, test_date=False, shipment_date=False):
+        table_suborders = self.base_selenium.get_table_rows(element='order:suborder_table')
+        required_suborder = self.base_selenium.get_row_cells_id_elements_related_to_header(row=table_suborders[sub_order_index],
+                                                                                             table_element='order:suborder_table')
+        response = {
+            "departments": "",
+            "material_types": "",
+            "article": "",
+            "test_plan": "",
+            "test_unit": "",
+            "test_date": "",
+            "shipment_date": ""
+        }
+        if departments :
+            response["departments"]="|".join(list(map(lambda s: str(s)[1:], required_suborder["departments"].text.split("\n")) ))
+        if test_plan :
+            response["test_plan"]="|".join(list(map(lambda s: str(s)[1:], required_suborder["testPlans"].text.split("\n")) ))
+        if test_unit :
+            response["test_unit"]="|".join(list(map(lambda s: str(s).split(' No:')[0][1:], required_suborder["testUnits"].text.split("\n")) ))
+        if articles :
+            response["article"]=required_suborder["article"].text.split("\n")[0]
+        if material_type :
+            response["material_types"]=required_suborder["materialType"].text.split("\n")[0]
+        if shipment_date :
+            pass
+        if test_date :
+            pass
+        return response
+
+    def remove_testplan_by_name(self, index, testplan_name):
+        suborder_table_rows = self.base_selenium.get_table_rows(element='order:suborder_table')
+        suborder_row = suborder_table_rows[index]
+        suborder_elements_dict = self.base_selenium.get_row_cells_id_elements_related_to_header(row=suborder_row,
+                                                                                            table_element='order:suborder_table')
+        self.base_selenium.update_item_value(item=suborder_elements_dict['testPlans'], item_text=testplan_name.replace("'", ''))
+
+    def update_suborder(self, sub_order_index=0, contacts=False, departments=[], material_type=False, articles=False, test_plans=[], test_units=[], shipment_date=False, test_date=False, form_view=True):
```

Comment:
1- Create a method for an update each item and call them inside this method like what we doing in creation
2- This method won't update as it won't delete the old values.

Reply:
about 1st point, okey will do
about the 2nd point, case of material type/ article, it will change the value because it is single select, and in case of departments, test plans, and test unit, i didn't want it to remove the old value because i already have another method for removing, i just needed it to add more values, or change the old values